framework/aster-frame/src/mm/space.rs

```diff
@@ -200,12 +238,29 @@ impl VmSpace {
     /// read-only. And both the VM space will take handles to the same
     /// physical memory pages.
     pub fn fork_copy_on_write(&self) -> Self {
+        let page_fault_handler = Once::new();
+        if let Some(handler) = self.page_fault_handler.get() {
+            page_fault_handler.call_once(|| handler.clone());
+        }
         let new_space = Self {
             pt: self.pt.fork_copy_on_write(),
+            page_fault_handler,
         };
         tlb_flush_all_excluding_global();
         new_space
     }
+
+    /// Gets a virtual memory environment that the `VmSpace` of current task
+    /// has been activated.
+    pub fn current() -> Option<CurrentVmSpace> {
+        let current_task = current_task()?;
+        let user_space = current_task.user_space()?;
+
+        user_space.vm_space().activate();
```

Comment:
Can current `VmSpace` not being activated? This may only happen when the first user task is not spawned yet. I think a solution may be letting the `aster_main` (in nix) being run as a kernel task that don't have a user space. Then it will fail before this check. Then we could debug assert activated instead of activating it.